communication/src/TlsTcpImpl.cpp

```diff
@@ -133,16 +133,16 @@ void TlsTCPCommunication::TlsTcpImpl::setReceiver(NodeNum _, IReceiver* receiver
   receiver_ = receiver;
 }
 
-int TlsTCPCommunication::TlsTcpImpl::sendAsyncMessage(const NodeNum destination, const char* msg, const size_t len) {
+int TlsTCPCommunication::TlsTcpImpl::sendAsyncMessage(const NodeNum destination, std::shared_ptr<OutgoingMsg> msg) {
```

Comment:
I think that `const std::shared_ptr<OutgoingMsg>& msg` is more appropriate.

Reply:
I think you mean std::shared_ptr<const OutgoingMsg>?

The idea here is that we take ownership of the memory in the interface (ICommunication). The buffer there is passed as rvalue reference and for the implementation we convert it to shared_ptr, so that we can use it for all sends.
Since we own this memory I think making it const is unnecessary limitation.